celery/schedules.py

```diff
@@ -101,6 +101,11 @@ def to_local(self, dt):
             return timezone.to_local_fallback(dt)
         return dt
 
+    def __eq__(self, other):
+        if isinstance(other, BaseSchedule):
+            return other.nowfun == self.nowfun
```

Comment:
I think there is a way, you can specify the full import path as a string and perform a dynamic import. This project performs lazy imports: https://github.com/Russell91/pythonpy/blob/master/pythonpy/__main__.py

This way you will have consistency on the function objects.